packages/framework-core/src/decorators/projection.ts

```diff
@@ -36,5 +36,5 @@ function registerProjection(originName: string, projectionMetadata: ProjectionMe
 }
 
 type ProjectionMethod<TEntity, TReadModel> = TypedPropertyDescriptor<
-  (_: TEntity, readModel?: TReadModel) => TReadModel | null
+  (_: TEntity, readModel?: TReadModel) => TReadModel | ReadModelAction
```

Comment:
One last suggestion: if you write here something like (maybe you need some parentesis): 
```typescript
type ProjectionMethod<TEntity, TReadModel> = TypedPropertyDescriptor<
     (_: TEntity, readModel?: TReadModel) => TReadModel | ReadModelAction  
  |  (_: TEntity, readModel?: TReadModel) => TReadModel
>
```

You would let users avoid writing the `| ReadModelAction` part in the projection signature in those projections that never delete a ReadModel. This way, the users' code would be cleaner 😃 

What do you think?

Reply:
that sounds great!! I was thinking about that yesterday but couldn't find a super quick solution for the templates. I think I will generate the templates (a.k.a `new read-model (...)`) without the `ReadModelAction` and let the user add it. Thank you!